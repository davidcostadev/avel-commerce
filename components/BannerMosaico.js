import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'
import styles from '../assets/scss/App.scss'
import theme from '../layout/theme'

const BannerMosaicoGrid = styled.div`
  display: flex;
  flex-wrap: wrap;
`
const BannerMosaicoCol = styled.div`
  flex: 1 1 100%;
  display: flex;
  justify-content: stretch;
  align-items: stretch;
  flex-direction: column;

  @media(min-width: 768px) {
    flex: 1 1 50%;
    max-width: 50%;
  }
`
const BannerMosaicoRow = styled.div`
  flex: 1 1 auto;
  display: flex;
  width: 100%;
  overflow: hidden;
`

const BannerText1 = styled.div`
  flex-basis: auto;
  color: ${theme.colorPrimary};
  font-size: 18px;
  margin-bottom: 20px;
  transition: transform 0.6s ease;
`
const BannerText2 = styled.div`
  flex-basis: auto;
  color: ${theme.white};
  font-size: 35px;
  margin-bottom: 20px;
  transition: transform 0.6s ease;
`
const BannerBtn = styled.a`
  flex-basis: auto;
  background-color: ${theme.colorPrimary};
  padding: 8px 12px;
  font-size: 18px;
  color: ${theme.white} !important;
  margin-bottom: 20px;
  border-width: 0;
  transition: background-color 0.4s ease;
  transition: transform 0.6s ease;

  &:hover {
    background-color: ${theme.colorPrimary}
  }
`

const BannerToCenter = `
  align-items: center;
  justify-content: center;
  ${BannerText1},
  ${BannerText2},
  ${BannerBtn} {
    text-align: center;
  }
`

const BannerToRight = `
  align-items: flex-end;
  ${BannerText1},
  ${BannerText2},
  ${BannerBtn} {
    text-align: right;
  }
`
const BannerToLeft = `
  align-items: flex-start;
  ${BannerText1},
  ${BannerText2},
  ${BannerBtn} {
    text-align: left;
  }
`

const BannerMosaicoItem = styled.div`
  flex: 1 1 50%;
  display: flex;
  flex-direction: column;
  background-size: cover;
  background-position: center;
  padding: 40px 60px;
  transition: transform 0.4s ease;

    &:hover {
    transform: scale(1.1);

    ${BannerText1},
    ${BannerText2},
    ${BannerBtn} {
      transform: scale(1.05)
    }
  }

  ${props => props.center ? BannerToCenter : ''}
  ${props => props.right ? BannerToRight : ''}
  ${props => props.left ? BannerToLeft : ''}
`


const BannerMosaico = () => (
  <BannerMosaicoGrid>
    <BannerMosaicoCol className={styles.bannerMosaicoCol}>
      <BannerMosaicoRow className={styles.bannerMosaicoRow}>
        <BannerMosaicoItem center className={`${styles.bannerMosaico} ${styles.bannerMosaicoCenter}`} style={{ backgroundImage: 'url(\'/static/img/phone.jpg\')' }}>
          <BannerText1 className={styles.bannerText1}>Celulares</BannerText1>
          <BannerText2 className={styles.bannerText2}>Tecnologia de Ponta</BannerText2>
          <Link href="/product">
            <BannerBtn className={styles.bannerBtn}>Ver Ofertas</BannerBtn>
          </Link>
        </BannerMosaicoItem>
      </BannerMosaicoRow>
    </BannerMosaicoCol>
    <BannerMosaicoCol className={styles.bannerMosaicoCol}>
      <BannerMosaicoRow className={styles.bannerMosaicoRow}>
        <BannerMosaicoItem left className={`${styles.bannerMosaico} ${styles.bannerMosaicoLeft}`} style={{ backgroundImage: 'url(\'/static/img/notebook.jpg\')' }}>
          <BannerText1 className={styles.bannerText1}>Notebook</BannerText1>
          <BannerText2 className={styles.bannerText2}>Ultima Geração</BannerText2>
          <Link href="/product">
            <BannerBtn className={styles.bannerBtn}>Ver Ofertas</BannerBtn>
          </Link>
        </BannerMosaicoItem>
      </BannerMosaicoRow>
      <BannerMosaicoRow className={styles.bannerMosaicoRow}>
        <BannerMosaicoItem right className={`${styles.bannerMosaico} ${styles.bannerMosaicoRight}`} style={{ backgroundImage: 'url(\'/static/img/perfume.jpg\')' }}>
          <BannerText1 className={styles.bannerText1}>Perfumes</BannerText1>
          <BannerText2 className={styles.bannerText2}>Melhores Marcas</BannerText2>
          <Link href="/product">
            <BannerBtn className={styles.bannerBtn}>Ver Ofertas</BannerBtn>
          </Link>
        </BannerMosaicoItem>
      </BannerMosaicoRow>
    </BannerMosaicoCol>
  </BannerMosaicoGrid>
)

export default BannerMosaico
