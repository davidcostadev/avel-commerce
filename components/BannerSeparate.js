import React from 'react'
import Link from 'next/link'
import styles from '../assets/scss/App.scss'

const BannerSeparate = () => (
  <div className={styles.bannerSeparateGrid}>
    <div className={styles.bannerSeparateCol}>
      <div className={styles.bannerSeparateRow}>
        <div className={`${styles.bannerSeparate} ${styles.bannerSeparateRight}`} style={{ backgroundImage: 'url(\'/static/img/camera.jpg\')' }}>
          <span className={styles.bannerText1}>Limpador de Tela</span>
          <span className={styles.bannerText2}>R$ 17,90</span>
          <Link href="/product">
            <a className={styles.bannerBtn}>Ver Detalhes</a>
          </Link>
        </div>
      </div>
    </div>
    <div className={styles.bannerSeparateCol}>
      <div className={styles.bannerSeparateRow}>
        <div className={`${styles.bannerSeparate} ${styles.bannerSeparateRight}`} style={{ backgroundImage: 'url(\'/static/img/mouse.jpg\')' }}>
          <span className={styles.bannerText1}>Mouse Mac</span>
          <span className={styles.bannerText2}>R$ 227,90</span>
          <Link href="/product">
            <a className={styles.bannerBtn}>Ver Detalhes</a>
          </Link>
        </div>
      </div>
    </div>
  </div>
)

export default BannerSeparate
