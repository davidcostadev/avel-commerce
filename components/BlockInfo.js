import React from 'react'
import PropTypes from 'prop-types'
import styles from '../assets/scss/App.scss'

const contentEscape = content => ({
  __html: content,
})

const BlockInfo = ({ content }) => (
  <section className={styles.blockInfo}>
    <header className={styles.blockInfoHeader}>
      <h2>Informações do Produto</h2>
    </header>
    <div className={styles.blockInfoContent} dangerouslySetInnerHTML={contentEscape(content)} />
  </section>
)

BlockInfo.propTypes = {
  content: PropTypes.string.isRequired,
}

export default BlockInfo
