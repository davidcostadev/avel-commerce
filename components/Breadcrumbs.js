import React from 'react'
import { Link } from '../routes'
import classNames from 'classnames'

import styles from '../assets/scss/App.scss'

const BreadcrumbItemLink = ({ item }) => (
  <li className={classNames([styles.breadcrumbItem, 'breadcrumb-item'])}>
    <Link route={item.route}><a>{item.title}</a></Link>
  </li>
)
const BreadcrumbItemCurrent = ({ item }) => (
  <li className={classNames([styles.breadcrumbItem, 'breadcrumb-item', 'active'])}>
    {item.title}
  </li>
)


const Breadcrumbs = ({ itens }) => (
  <ol className={classNames([styles.breadcrumb, 'breadcrumb'])}>
    {itens.map((item, index) => {
      console.log(itens.length - 1 === index)
      if (itens.length - 1 === index) {
        return <BreadcrumbItemCurrent item={item} key={index}/>
      }

      return <BreadcrumbItemLink item={item} key={index} />
    })}
  </ol>
)

export default Breadcrumbs
