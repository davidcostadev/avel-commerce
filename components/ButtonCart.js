import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Link } from '../routes'
import styles from '../assets/scss/App.scss'
import theme from '../layout/theme'

const Badge = styled.span`
  display: inline-block;
  width: 22px;
  height: 22px;
  padding: 5px 0px;
  background: ${theme.colorSecond};
  border-radius: 50%;
  font-size: 12px;
  vertical-align: super;
  color: white;
  transform: translate(4px, -8px);
`

const ButtonCartWrap = styled.a`
  display: flex;
  align-items: center;
  color: ${theme.colorPrimary} !important;
  background-color: white;
  border-radius: 3px;
  border-width: 0;
  padding: 0 10px;
  cursor: pointer;

  &:focus,
  &:hover {
    background-color: ${theme.buttonDefault} !important;
    color: green;
  }
  &:active {
    background-color: ${theme.buttonDefaultHover} !important;
  }
  i {
    font-size: 32px;
    vertical-align: middle;
  }
`

export const ButtonCart = ({ quant }) => (
  <Link route="/cart">
    <ButtonCartWrap>
      <i className="ion-ios-cart-outline" />
      <Badge className={`badge ${styles.badge}`}>{quant}</Badge>
    </ButtonCartWrap>
  </Link>
)

const ButtonCartStyled = styled(ButtonCart)`

`


ButtonCart.propTypes = {
  quant: PropTypes.number,
}

ButtonCart.defaultProps = {
  quant: 0,
}

const mapState = state => ({
  quant: state.cartItens.length,
})

export default connect(mapState)(ButtonCartStyled)
