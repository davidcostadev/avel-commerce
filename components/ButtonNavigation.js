import React from 'react'
import { connect } from 'react-redux'
import classNames from 'classnames'
import PropsTypes from 'prop-types'
import { toggleMenu } from '../flux/config/configActions'

import {
  MenuButton,
} from '../layout/Header'

const ButtonNavigation = ({ menuMobile, toogleMenu }) => {
  const classList = {
    'ion-navicon': !menuMobile,
    'ion-ios-close-empty': menuMobile,
  }

  return (
    <MenuButton onClick={toogleMenu}>
      <i className={classNames(classList)} />
    </MenuButton>
  )
}

ButtonNavigation.propTypes = {
  menuMobile: PropsTypes.bool.isRequired,
  toogleMenu: PropsTypes.func.isRequired,
}

const mapState = state => state

const mapDispatchToProps = dispatch => ({
  toogleMenu: () => dispatch(toggleMenu()),
})

export default connect(mapState, mapDispatchToProps)(ButtonNavigation)
