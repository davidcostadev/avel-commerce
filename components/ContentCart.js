import React from 'react'
// import PropTypes from 'prop-types'
import { floatToReal, StringToReal } from '../utils/money'
import ProductCart from '../components/ProductCart'
import Table from '../layout/Table'
import * as Cart from '../layout/Cart'

const Products = ({ products, changeQuant }) => {
  if (!products.length) {
    return (
      <tr><td colSpan="5">Nenhum produto no carrinho</td></tr>
    )
  }

  return products.map(product => (
    <ProductCart key={product.PS_ID_PRODUTO} product={product} changeQuant={changeQuant} />
  ))
}

const Checkout = ({ cart, cartItens }) => {
  if (!cartItens.length) {
    return null
  }

  return (
    <Cart.CheckoutRow>
      <Cart.CalcShipping>
        <p>Consulte o prazo de entrega e o frete para seu CEP:</p>
        <div className="form-group">
          <div className="input-group">
            <input type="text" className="form-control" placeholder="00000-000" />
            <span className="input-group-btn">
              <button className="btn btn-default">Calcular</button>
            </span>
          </div>
        </div>
        <div className="form-check">
          <label htmlFor="" className="form-check-label">
            <input type="radio" className="form-check-input" />
            Retirar da loja(sem frete) - Em Média 0 dia(s) úteis - R$ 0,00
        </label>
          <label htmlFor="" className="form-check-label">
            <input type="radio" className="form-check-input" />
            Retirar da loja(sem frete) - Em Média 0 dia(s) úteis - R$ 0,00
        </label>
          <label htmlFor="" className="form-check-label">
            <input type="radio" className="form-check-input" />
            Retirar da loja(sem frete) - Em Média 0 dia(s) úteis - R$ 0,00
        </label>
        </div>
      </Cart.CalcShipping>
      <div className="card">
        <div className="card-body">
          <p className="card-text">
            <span>Subtotal</span>
            <span>R$ {StringToReal(cart.PS_VL_SUBTOTAL)}</span>
          </p>
          <p className="card-text">
            <span>Frete</span>
            <span>R$ {floatToReal(0)}</span>
          </p>
          <p className="card-text">
            <span><strong>Total</strong></span>
            <span>R$ {StringToReal(cart.PS_VL_TOTAL_GERAL)}</span>
          </p>
          <button className="btn btn-primary">Finalizar Comprar</button>
        </div>
      </div>
    </Cart.CheckoutRow>
  )
}


const ContentCart = ({ cart, cartItens, changeQuant }) => {
  return (
    <Cart.Page>
      <Cart.Title>Carrinho</Cart.Title>
      <Table>
        <thead>
          <tr>
            <th />
            <th>Produto</th>
            <th>Preço Unit.</th>
            <th>Quantidade</th>
            <th>Subtotal</th>
          </tr>
        </thead>
        <tbody>
          <Products products={cartItens} changeQuant={changeQuant} />
        </tbody>
      </Table>
      <Checkout cart={cart} cartItens={cartItens} />
    </Cart.Page>
  )
}

export default ContentCart
