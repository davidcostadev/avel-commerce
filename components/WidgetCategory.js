import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import theme from '../layout/theme'
import sortAlphabetfrom from '../utils/sort'

export const LinkItem = ({ item }) => (
  <li>
    <a href={item.path}>{item.title}</a>
  </li>
)

LinkItem.propTypes = {
  item: PropTypes.object.isRequired,
}


const Widget = styled.div`
  &+& {
    margin-top: 30px;
  }
`
const WidgetTitle = styled.h3`
  font-size: 14px;
  text-align: center;
  padding: 5px 0;
  margin-bottom: 0;
  background: ${theme.gray300};
  color: ${theme.gray500}
  `
// const WidgetContent = styled.div`
//   background: ${theme.white};
//   padding: 15px;
// `
const WidgetCategories = styled.ul`
  margin-bottom: 0;
  padding-left: 0;
  list-style: none;
  display: flex;
  flex-direction: column;

  li {
    display: flex;
  }
  a {
    flex: 1;
    padding: 8px 16px;
    background-color: ${theme.white};

    &:hover {
      background-color: ${theme.colorPrimary};
      color: ${theme.white};
    }

    &:active {
      background-color: ${theme.colorPrimary};
      color: ${theme.white};
    }
  }
`


export const WidgetCategory = ({ title, menu }) => (
  <Widget>
    <WidgetTitle>{title}</WidgetTitle>
    <WidgetCategories>
      {menu.sort(sortAlphabetfrom).map(item => (
        <LinkItem key={item.path} item={item} />
      ))}
    </WidgetCategories>
  </Widget>
)

WidgetCategory.propTypes = {
  title: PropTypes.string.isRequired,
  menu: PropTypes.array.isRequired,
}

export default WidgetCategory
