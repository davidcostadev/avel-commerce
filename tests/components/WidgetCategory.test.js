/* eslint no-undef: "off" */
import React from 'react'
import renderer from 'react-test-renderer'
import { LinkItem, WidgetCategory } from '../../components/WidgetCategory'

describe('Component <WidgetCategory />', () => {
  it('<LinkItem />', () => {
    const tree = renderer.create((
      <LinkItem
        item={{ path: '#', title: 'title' }}
      />
    )).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('<WidgetCategory />', () => {
    const menu = [
      {
        path: '#1',
        title: 'title 1',
      },
      {
        path: '#2',
        title: 'title 2',
      },
    ]
    const tree = renderer.create((
      <WidgetCategory
        title="Titulo Widget"
        menu={menu}
      />
    )).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
