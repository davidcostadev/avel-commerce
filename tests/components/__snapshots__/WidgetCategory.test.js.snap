// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <WidgetCategory /> <LinkItem /> 1`] = `
<li>
  <a
    href="#"
  >
    title
  </a>
</li>
`;

exports[`Component <WidgetCategory /> <WidgetCategory /> 1`] = `
<div
  className="WidgetCategory__Widget-s7sa3mi-0 hLfuzK"
>
  <h3
    className="WidgetCategory__WidgetTitle-s7sa3mi-1 kHkFYp"
  >
    Titulo Widget
  </h3>
  <ul
    className="WidgetCategory__WidgetCategories-s7sa3mi-2 dKwjNk"
  >
    <li>
      <a
        href="#1"
      >
        title 1
      </a>
    </li>
    <li>
      <a
        href="#2"
      >
        title 2
      </a>
    </li>
  </ul>
</div>
`;
